XSConsoleData.py

```diff
@@ -980,21 +980,58 @@ def ManagementIP(self, inDefault = None):
 
     def ManagementNetmask(self, inDefault = None):
         retVal = inDefault
-
         for pif in self.derived.managementpifs([]):
-            retVal = pif['netmask']
-            if retVal:
-                break
+            ipv6 = pif['primary_address_type'].lower() == 'ipv6'
+            configuration_mode = pif['ipv6_configuration_mode'] if ipv6 else pif['ip_configuration_mode']
+            if configuration_mode.lower().startswith('static'):
+                # For static IP the API address is correct
+                retVal = pif['IPv6'][0].split('/')[1] if ipv6 else pif['netmask']
+            elif configuration_mode.lower().startswith('dhcp'):
+                # For DHCP,  find the gateway address by parsing the output from the 'route' command
+                if 'bridge' in pif['network']:
+                    device = pif['network']['bridge']
+                else:
+                    device = pif['device']
+
+                device = ShellUtils.MakeSafeParam(device)
+
+                ipre = r'[0-9a-f.:]+'
+                ifRE = re.compile(r'\s*inet\s+' + ipre + '\s+netmask\s+(' + ipre + r')\s+broadcast\s+(' + ipre + r')\s*$',
+                        re.IGNORECASE)
+
+                ifconfig = commands.getoutput("/sbin/ifconfig '"+device+"'").split("\n")
+                for line in ifconfig:
+                    match = ifRE.match(line)
+                    if match:
+                        retVal = match.group(1)
+                        break
 
```

Comment:
This code is no longer relevant Whatever the mode the netmask should be read from xapi.